kotlin-mbedtls/src/main/kotlin/org/opencoap/ssl/transport/DtlsSessionContext.kt

```diff
@@ -20,10 +20,34 @@ typealias AuthenticationContext = Map<String, String>
 
 data class DtlsSessionContext @JvmOverloads constructor(
     val authenticationContext: AuthenticationContext = emptyMap(),
-    val peerCertificateSubject: String? = null
+    val peerCertificateSubject: String? = null,
+    val cid: ByteArray? = null
 ) {
     companion object {
         @JvmField
         val EMPTY = DtlsSessionContext()
     }
+
+    override fun equals(other: Any?): Boolean {
```

Comment:
Why default `equals` and `hashCode` implementations are not enough?

Reply:
yeah the ByteArray doesn't work with data class without these for unknown reason.